### PR TITLE
case sensetive version of sql member type

### PR DESCRIPTION
**Pull request by alibazregar**

### CODE REVIEW

File: renderEngine/Command/Source/BaseClasses/SqlMember.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        context.waitToGetSourceAsync(dataMemberName)
+      );
+      const sources = await Promise.all(getSourcePromises);
+      const insertPromises = sources.map(async (source, index) => {
+        const dataArray = source.data;
+        const dataMemberName = dataMemberNames[index];
+        const createTableSql = `CREATE TABLE [${dataMemberName}] (${Object.keys(
+          dataArray[0]
+        )
+          .map((key) => `${key} STRING`)
```
Comment (Qamsari) [inline renderEngine/Command/Source/BaseClasses/SqlMember.js:33]: we have columns property in [IDataSource  ](https://github.com/Manzoomeh/BasisCore.Server.Node/blob/d6ecfc32c0fe024ce3ead6e97b9e031da6b4a74a/renderEngine/Source/IDataSource.js#L9C6-L9C16)
```javascript
/**@type {Array<string>} */
  get columns() {
    return this.data?.length > 0 ? Object.keys(this.data[0]) : [];
  }
```

use it please.

Comment (Qamsari) [reply renderEngine/Command/Source/BaseClasses/SqlMember.js:33]: also all column is not string. use `STRING` for all maybe cause error. please add test for case like `join over number` column

File: renderEngine/Command/Source/BaseClasses/SqlMember.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          )
+            .map((val) => `'${val}'`)
+            .join(", ")})`;
+          return this.executeQueryAsync(db, insertSql);
+        });
+        await Promise.all(insertPromises);
+      });
+      await Promise.all(insertPromises);
+      const queryResult = await this.executeQueryAsync(db, content);
+      db = null;
```
Comment (Qamsari) [inline renderEngine/Command/Source/BaseClasses/SqlMember.js:48]: set to null is sufficient?!

File: renderEngine/Command/Source/BaseClasses/SqlMember.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      return new JsonSource(queryResult, this.name);
+    } catch (err) {
+      console.log(err);
```
Comment (Qamsari) [inline renderEngine/Command/Source/BaseClasses/SqlMember.js:51]: return error for generate result

File: renderEngine/Context/RequestContext.js [hunk 4]
```diff
@@ -17,7 +17,7 @@ export default class RequestContext extends ContextBase {
    * @param {IRoutingRequest} request
    */
   constructor(settings, request) {
-    super(null, Number(request.cms?.dmnid));
+    super(null, Number(request?.cms?.dmnid));
```
Comment (Qamsari) [inline renderEngine/Context/RequestContext.js:20]: when `request` is null?!

File: test/command/InlineSource/inlinesourceSqlCommand.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
+import alasql from "alasql";
+import ServiceSettings from "../../../models/ServiceSettings.js";
+import CancellationToken from "../../../renderEngine/Cancellation/CancellationToken.js";
+import GroupCommand from "../../../renderEngine/Command/Collection/GroupCommand.js";
+import RequestContext from "../../../renderEngine/Context/RequestContext.js";
+var setting = new ServiceSettings({});
+const context = new RequestContext(setting);
```
Comment (Qamsari) [inline test/command/InlineSource/inlinesourceSqlCommand.js:7]: `TestContext ` is not better?!

**Review by Qamsari — COMMENTED** [review]
please fix my issue

File: renderEngine/Command/Source/BaseClasses/SqlMember.js (hunk 2 above)
Comment (alibazregar) [reply renderEngine/Command/Source/BaseClasses/SqlMember.js:48]: yes it is works in alasql

File: renderEngine/Command/Source/BaseClasses/SqlMember.js (hunk 1 above)
Comment (alibazregar) [reply renderEngine/Command/Source/BaseClasses/SqlMember.js:33]: i added that ; thanks.

**Pull request merged**